snap/quota/quota.go

```diff
@@ -271,11 +272,44 @@ func (grp *Group) getAllowedCPUs() []int {
 	return grp.CPULimit.AllowedCPUs
 }
 
+// GetCPUSetQuota returns the currently active CPU set quota for this group, which
+// could originate from a parent
+func (grp *Group) GetCPUSetQuota() []int {
+	if grp.CPULimit != nil && len(grp.CPULimit.AllowedCPUs) != 0 {
+		return grp.CPULimit.AllowedCPUs
+	}
+
+	parent := grp.parentGroup
+	for parent != nil {
+		if parent.CPULimit != nil && len(parent.CPULimit.AllowedCPUs) != 0 {
+			return parent.CPULimit.AllowedCPUs
+		}
+		parent = parent.parentGroup
+	}
+	return []int{}
+}
+
+// GetCorrectedCPUCount returns the maximum number of allowed CPU cores for
+// this group. It needs to take into account that CPU set might have been set
+// to limit the number of cores, or a direct limit on the number of cores.
+// Goal is to select the most restrictive limit.
+func (grp *Group) GetCorrectedCPUCount() int {
+	cpuCount := runtime.NumCPU()
+	cpuSetCount := len(grp.GetCPUSetQuota())
+	if cpuSetCount != 0 && cpuSetCount < cpuCount {
+		cpuCount = cpuSetCount
+	}
+	if grp.CPULimit != nil && grp.CPULimit.Count != 0 && grp.CPULimit.Count < cpuCount {
```

Comment:
Don't we need to inspect the parent groups, in case this CPULimit is unset on this one?

Reply:
Really good question, but this is actually the intended behaviour, I have added documentation explaining this

Reply:
And updated the branch to newest master!

Reply:
Thanks!